### PR TITLE
Fix to use localtime for Linux

### PR DESCRIPTION
**Pull request by kwoodle**
Changed axis_type.cpp to use localtime since localtime_s doesn't work in Linux. Added     add_definitions(-D_CRT_SECURE_NO_WARNINGS) to source/matplot/CMakeLists.txt to make localhost work on Windows. I usually work in Linux. Not sure if this breaks anything on Windows, although I did build ContinuousBuild target and ran some examples.

### CODE REVIEW

File: source/matplot/CMakeLists.txt [hunk 1]
```diff
@@ -159,6 +159,9 @@ endif ()
 #######################################################
 ### Library options                                 ###
 #######################################################
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
```
Comment (alandefreitas) [inline source/matplot/CMakeLists.txt:163]: target_compile_definitions?

File: source/matplot/core/axis_type.cpp [hunk 2]
```diff
@@ -256,9 +256,9 @@ namespace matplot {
                 if (is_timestamp_) {
                     char buff[20] = {0};
                     time_t now = (time_t)tick_values_[i];
-                    struct tm buf;
-                    localtime_s(&buf, &now);
-                    strftime(buff, 20, tick_label_format_.c_str(), &buf);
+                    struct tm* buf;
```
Comment (alandefreitas) [inline source/matplot/core/axis_type.cpp:259]: Isn't `#ifdef WIN32 ...` simpler than the external `-D_CRT_SECURE_NO_WARNINGS` here?

**Review by alandefreitas — COMMENTED** [review]
Great. We could squash this into a single commit.

File: source/matplot/core/axis_type.cpp (hunk 2 above)
Comment (kwoodle) [reply source/matplot/core/axis_type.cpp]: I've added macros and thread-safe localtime_r to axis_type.cpp, removed the change to CMakeLists.txt, and pushed the changes to my fork. Do I need to do anything else to integrate the changes? Also, when I run the .exe examples on Windows, gnuplot has trouble reading input. Hope you can check this. Thanks for your suggestions.

File: source/matplot/core/axis_type.cpp [hunk 3]
```diff
@@ -257,7 +257,11 @@ namespace matplot {
                     char buff[20] = {0};
                     time_t now = (time_t)tick_values_[i];
                     struct tm buf;
+#if defined(__unix__)
+                    localtime_r(&now, &buf);
+#elif defined(_WIN32)
                     localtime_s(&buf, &now);
+#endif
```
Comment (alandefreitas) [inline source/matplot/core/axis_type.cpp:264]: Nice. We need a third alternative when both conditions are false.

```cpp
#if defined(__unix__)
                    localtime_r(&now, &buf);
#elif defined(_WIN32)
                    localtime_s(&buf, &now);
#else
                    // use std::localtime from https://en.cppreference.com/w/cpp/chrono/c/localtime
#endif
```

We also need an equivalent pattern to include the proper headers:

```cpp
#if defined(__unix__)
#include <header with localtime_r>
#elif defined(_WIN32)
#include <header with localtime_s>
#else
#include <ctime>
#endif
```

File: source/matplot/CMakeLists.txt [hunk 4]
```diff
@@ -159,7 +159,6 @@ endif ()
 #######################################################
 ### Library options                                 ###
 #######################################################
-
```
Comment (alandefreitas) [inline source/matplot/CMakeLists.txt:162]: Can't you just rollback this non-change?

File: source/matplot/core/axis_type.cpp (hunk 2 above)
Comment (alandefreitas) [reply source/matplot/core/axis_type.cpp]: > I've added macros and thread-safe localtime_r to axis_type.cpp, removed the change to CMakeLists.txt, and pushed the changes to my fork. 

Nice. Could you squash these commits?

> Do I need to do anything else to integrate the changes? 

I think we are missing the `std::localtime` default now.

> Also, when I run the .exe examples on Windows, gnuplot has trouble reading input. Hope you can check this. 

Sure. That's unrelated but I'll have a look.

Thanks for the PR.

File: source/matplot/core/axis_type.cpp [hunk 5]
```diff
@@ -257,7 +263,13 @@ namespace matplot {
                     char buff[20] = {0};
                     time_t now = (time_t)tick_values_[i];
                     struct tm buf;
+#if defined(__unix__)
+                    localtime_r(&now, &buf);
+#elif defined(_WIN32)
                     localtime_s(&buf, &now);
+#else
+                    buf = std::localtime(&now);
```
Comment (alandefreitas) [inline source/matplot/core/axis_type.cpp:271]: We have a bug here:

https://github.com/alandefreitas/matplotplusplus/runs/7770952743?check_suite_focus=true#step:5:20

**Comment by alandefreitas** [comment]
Nice! Thanks!

**Comment by kwoodle** [comment]
You're too kind!

**Pull request merged**